core/frontend/src/views/NetworkTestView.vue

```diff
@@ -0,0 +1,227 @@
+<template>
+  <v-container
+    class="d-flex justify-center fluid"
+  >
+    <v-card>
+      <v-card-title class="justify-center">
+        Network speed and latency test
+      </v-card-title>
+
+      <v-divider />
+
+      <v-col>
+        <v-col
+          class="d-flex justify-center"
+        >
+          <v-progress-circular
+            :rotate="90"
+            :width="15"
+            :value="speed * 100 / max_speed"
+            ratio="1"
+            color="blue"
+            size="200"
+          >
+            {{ speed.toFixed(2) }} MB/s
+          </v-progress-circular>
+
+          <v-list>
+            <v-list-item>
+              <v-list-item-icon>
+                <v-icon>
+                  mdi-download
+                </v-icon>
+              </v-list-item-icon>
+              <v-list-item-subtitle>
+                {{ download_speed ? download_speed.toFixed(2) : '...' }} MB/s
+              </v-list-item-subtitle>
+            </v-list-item>
+
+            <v-list-item>
+              <v-list-item-icon>
+                <v-icon>
+                  mdi-upload
+                </v-icon>
+              </v-list-item-icon>
+              <v-list-item-subtitle>
+                {{ upload_speed ? upload_speed.toFixed(2) : '...' }} MB/s
+              </v-list-item-subtitle>
+            </v-list-item>
+
+            <v-list-item>
+              <v-list-item-icon>
+                <v-icon>
+                  mdi-timer-outline
+                </v-icon>
+              </v-list-item-icon>
+              <v-list-item-subtitle>
+                {{ latency_us ? latency_us.toFixed(0) : '...' }} us
+              </v-list-item-subtitle>
+            </v-list-item>
+          </v-list>
+        </v-col>
+        <div
+          class="transition-swing pa-6"
+          align="center"
+          v-text="state"
+        />
+        <v-card-actions class="justify-center">
+          <v-btn
+            @click="start()"
+          >
+            <v-icon color="green">
+              mdi-play
+            </v-icon>
+          </v-btn>
+        </v-card-actions>
+      </v-col>
+    </v-card>
+  </v-container>
+</template>
+
+<script lang="ts">
+import Vue from 'vue'
+import { differenceInSeconds } from 'date-fns'
+
+import notifications from '@/store/notifications'
+import { network_speed_test_service } from '@/types/frontend_services'
+import back_axios from '@/utils/api'
+
+enum State {
+  None = '...',
+  DownloadSpeed = 'Download speed',
+  UploadSpeed = 'Upload speed',
+  Done = 'Done',
```

Comment:
1. Click to test download/upload speed
2. Testing download speed
3. Testing upload speed
4. Test done